### PR TITLE
build: sveltekit breaking changes

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

SvelteKit introduced a new syntax for its routing, as well as for its data loading, in the newest pre-1.0 release.
This PR performs the migration.

# Resources

- Migration guide: https://github.com/sveltejs/kit/discussions/5774
- Tutorial: https://joshcollinsworth.com/blog/sveltekit-breaking-changes

# Changes

- bump all dependencies
- migrate routes
- migrate package script
- adapt configuration

### CODE REVIEW

**Comment by peterpeterparker** [comment]
Blocked. The [certified assets canister](https://github.com/dfinity/certified-assets/) does not support SvelteKit routing anymore.

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -3,7 +3,7 @@
   import DocsMenu from "$docs/components/DocsMenu.svelte";
   import { page } from "$app/stores";
   import { goto, afterNavigate } from "$app/navigation";
-  import HeaderTitle from "../lib/components/HeaderTitle.svelte";
```
Comment (lmuntaner) [inline src/routes/+layout.svelte:6]: oh, is this some new cool svelte magic?

File: vite.config.js [hunk 2]
```diff
@@ -7,7 +7,6 @@ const config = {
   resolve: {
     alias: {
       $docs: resolve("./src/docs"),
-      $lib: resolve("./src/lib"),
```
Comment (lmuntaner) [inline vite.config.js:10]: why is one removed and the other stays?

**Review by lmuntaner — APPROVED** [review]
A couple of comments, already approving.

File: src/routes/+layout.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/routes/+layout.svelte:6]: you mean the `$lib`? not new, I think you can do that with rollup or any bundler - i.e. we can do that in NNS-dapp too if we want too but we would need an external lib. In SvelteKit it's included and configured out of the box.

File: vite.config.js (hunk 2 above)
Comment (peterpeterparker) [reply vite.config.js:10]: before last version of SvelteKit/Vite we had to overwrite the default `$lib` to be able to declare `$docs` as well but, after the migration, it's the contrary, `$lib` should not be re-declared as available in any case. I guess the custom config used to overwrite the default config while now they merge it.

Comment (lmuntaner) [reply vite.config.js:10]: ok, thanks for the explanation!

File: src/routes/+layout.svelte (hunk 1 above)
Comment (lmuntaner) [reply src/routes/+layout.svelte:6]: ah, ok, let's think about it. For me it would depend more on the VS Code integration to be honest.

**Pull request merged**